grakn-graph/src/main/java/ai/grakn/graph/internal/InstanceImpl.java

```diff
@@ -88,26 +92,20 @@
      */
     @Override
     public void delete() {
-        InstanceImpl<?, ?> parent = this;
-        Set<CastingImpl> castings = parent.castings();
+        Set<Relation> relations = castingsInstance().map(Casting::getRelation).collect(Collectors.toSet());
```

Comment:
Yep which in turn iterates over the instances of those relations. So if one thing is deleted in the wrong order you get a null pointer.